### ISSUE TITLE
Use a CDK deploy role

### ISSUE THREAD

**Issue by Rabadash8820** [DerploidEntertainment/Website#65]
Right now, our CDK GitHub Actions job is just using my IAM account credentials to deploy CloudFormation Stacks. Instead, it should use a dedicated role, to which GitHub Actions is granted "assume" permissions. See [the docs](https://confluence.agsoc-dev.agscollab.com/pages/viewpage.action?pageId=241216419) for more info.

**Comment by Rabadash8820** [comment]
Looks I linked to some internal Amazon page by accident in the description. 🤔

Googling around now, I can't actually find any official, up-to-date docs about "deploy roles". The [Deploying AWS CDK Applications](https://docs.aws.amazon.com/cdk/v2/guide/deploy.html#deploy-how-deploy) page says:

> the IAM roles created in your AWS environment during bootstrapping are used to establish permissions to perform the actions needed for deployment.

The [CDK `deploy`](https://docs.aws.amazon.com/cdk/v2/guide/ref-cli-cmd-deploy.html) command docs don't have anything about roles (other than during hotswaps, which we haven't needed yet). The [security best practices](https://docs.aws.amazon.com/cdk/v2/guide/best-practices-security.html#best-practices-security-permissions) clarify that whoever runs the CDK CLI basically assumes a `DeploymentActionRole` defined in the bootstrap stack, which provides permission to call Cloud Formation, which then uses a `CloudFormationExecutionRole` to actually create/modify resources, which has `AdministratorAccess` permissions. So the main ways that we could limit permissions are to restrict who has access to the `DeploymentActionRole` and restrict the permissions of `CloudFormationExecutionRole`. Continuously updating the `CloudFormationExecutionRole` to have the least possible privileges sounds like a super maintenance headache though, and lots of people online seem to agree (see [this Reddit thread](https://www.reddit.com/r/aws/comments/hwc4jf/iam_role_for_automatic_aws_cdk_deployment/), e.g.), so I think I'm content to leave it with admin privileges. And considering that my IAM User is the only one defined in the AWS account, restricting who can assume the `DeploymentActionRole` is pointless. 

**TL;DR** Our current security posture seems fine; no new roles are needed. 👍
